api/v1/SEL.php

```diff
@@ -652,19 +652,21 @@ public function GetHeatmapSettings(): array
         $riskMapSettings = $configValues["risk_heatmap_settings"];
 
         $restrictionLayerExceptions = array();
-        foreach ($riskMapSettings["restriction_layer_exceptions"] as $data) {
-            $layerData = $this->getDatabase()->query(
-                "SELECT layer_id FROM layer WHERE layer_name = ?",
-                array($data)
-            );
-            if (count($layerData) > 0) {
-                $restrictionLayerExceptions[] = $layerData[0]["layer_id"];
-            } else {
-                $this->getLogger()->serverEvent(
-                    "SEL_API",
-                    Log::WARNING,
-                    "Unknown layer with name ".$data." found in SEL Configuration file"
+        if (!empty($riskMapSettings["restriction_layer_exceptions"])) {
+            foreach ($riskMapSettings["restriction_layer_exceptions"] as $data) {
+                $layerData = $this->getDatabase()->query(
+                    "SELECT layer_id FROM layer WHERE layer_name = ?",
+                    array($data)
                 );
+                if (count($layerData) > 0) {
+                    $restrictionLayerExceptions[] = $layerData[0]["layer_id"];
+                } else {
+                    $this->getLogger()->serverEvent(
+                        "SEL_API",
+                        Log::WARNING,
+                        "Unknown layer with name ".$data." found in SEL Configuration file"
+                    );
+                }
```

Comment:
bij voorkeur zo. om onnodig inspringen te voorkomen. Maar ik ga code niet afkeuren om "stijl". Maar ik wijs je er wel op ;)

```suggestion
                if (count($layerData) > 0) {
                    $restrictionLayerExceptions[] = $layerData[0]["layer_id"];
                    continue;
                }
                $this->getLogger()->serverEvent(
                    "SEL_API",
                    Log::WARNING,
                    "Unknown layer with name ".$data." found in SEL Configuration file"
                );
```

Reply:
heel terecht! ik pas het aan